pkg/kubeconfig/kubeconfig.go

```diff
@@ -126,19 +143,14 @@ func (cfg *configLoader) GetConfig() error {
 		if err != nil {
 			return err
 		}
+	} else {
+		existingKubeConfig, err = clientcmd.LoadFromFile(localPath)
+		if err != nil {
+			return err
+		}
 	}
 
-	f, err := os.OpenFile(localPath, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
-	if err != nil {
-		return err
-	}
-	_, err = f.WriteString(string(bytes))
-	if err != nil {
-		return err
-	}
-	defer f.Close()
-	logger.Always("Wrote kubeconfig to [%s]", localPath)
-	return nil
+	return clientcmd.WriteToFile(*mergeKubeconfigs([]*clientcmdapi.Config{existingKubeConfig, newKubeConfig}), localPath)
```

Comment:
I'm wondering could we improve this. If the `WriteToFile` function takes struct as it is, and not a pointer to it. maybe we can just make `mergeKubeconfigs` return struct instead of a pointer.

Potentially the struct is not that big, and we're not modifying it all, so that could work and be okay performance wise.

Not a big thing, it can stay both ways, but looking can we make this more readable. :)

Reply:
I wanted to keep the return type of `mergeKubeconfigs` consistent with that of `clientcmdapi.NewConfig()`. This is also consistent with most `clientcmd` loader functions.
The pattern I've seen here is, functions return a pointer but take parameters by-value. For that reason, if it is OK with you, I'll keep this as is.

Reply:
Although, I agree with you on the readability of this code and I have made it more readable. LMK WYT.

Reply:
Right, that makes a lot of sense. I like how it looks right now, so let's leave it as it is.